Sources/FoundationMacros/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+# SwiftSyntax Dependency
+include(FetchContent)
+find_package(SwiftSyntax)
+if(NOT SwiftSyntax_FOUND)
+    # If building at desk, check out and link against the SwiftSyntax repo's targets
+    FetchContent_Declare(SwiftSyntax
+        GIT_REPOSITORY https://github.com/apple/swift-syntax.git
+        GIT_TAG 510.0.0)
+    FetchContent_MakeAvailable(SwiftSyntax)
+endif()
+
+add_library(FoundationMacros SHARED
+    FoundationMacros.swift
+    PredicateMacro.swift)
+    
+target_compile_definitions(FoundationMacros PRIVATE FOUNDATION_MACROS_LIBRARY)
+
+set_target_properties(FoundationMacros
+    PROPERTIES
+        ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib
+        INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_BINARY_DIR}/lib
+)
+
+target_link_libraries(FoundationMacros PUBLIC
+    SwiftSyntax::SwiftSyntax
```

Comment:
The 510.x tag of SwiftSyntax doesn't namespace its targets like this.
Newer versions will though.